src/main/java/com/mongodb/kafka/connect/source/MongoSourceConfig.java

```diff
@@ -184,15 +184,70 @@ public class MongoSourceConfig extends AbstractConfig {
   private static final String TOPIC_NAMESPACE_MAP_DISPLAY = "The namespace to topic map";
   private static final String TOPIC_NAMESPACE_MAP_DOC =
-      "A json map that maps change stream document namespaces to topics.\n"
-          + "For example: `{\"db\": \"dbTopic\", \"db.coll\": \"dbCollTopic\"}` will map all "
-          + "change stream documents from the `db` database to `dbTopic.<collectionName>` apart from"
-          + "any documents from the `db.coll` namespace which map to the `dbCollTopic` topic.\n"
-          + "If you want to map all messages to a single topic use `*`: "
-          + "For example: `{\"*\": \"everyThingTopic\", \"db.coll\": \"exceptionToTheRuleTopic\"}` "
-          + "will map all change stream documents to the `everyThingTopic` apart from the `db.coll` "
-          + "messages."
-          + "Note: Any prefix and suffix configuration will still apply.";
+      "A JSON object configuring how to map a MongoDB change stream document namespace"
+          + "(a database name optionally concatenated with a collection name, separated by full stop '.')"
+          + " to a Kafka topic name. Used by the `DefaultTopicMapper`."
+          + "\nThe name in each pair is a namespace pattern, the value is a string representing the corresponding topic name template."
+          + " Pairs are ordered. When there are multiple pairs with equal namespace patterns (duplicates),"
+          + " they are deduplicated and only one pair is taken into account:"
+          + " its position is taken from the first pair among duplicates,"
+          + " its topic name template is taken from the last pair among duplicates."
+          + " After deduplication, pairs with an empty topic name template are ignored when computing topic names."
+          + " Note that a topic name computed based on this configuration is then decorated"
+          + " using the `topic.prefix` and `topic.suffix` configuration properties."
+          + "\nThere are two kinds of pairs:"
+          + "\n- Simple. The namespace pattern must not contain solidus '/' and can be either of the following:"
+          + "\n  - A namespace with a collection name, in which case it matches only that namespace."
+          + " The topic name template is interpreted as the topic name."
+          + "\n  - A namespace with only a database name, in which case it matches any namespace having that database name."
+          + " The matching namespace may either have a collection name, or not:"
+          + "\n    - If there is a collection name, then the topic name is computed"
+          + " by concatenating the topic name template and the collection name from the matching namespace, separated by `topic.separator`."
+          + "\n    - If there is no collection name, then the the topic name template is interpreted as the topic name."
+          + "\n- Regex. The namespace pattern starts with solidus '/',"
```

Comment:
> Regex. The namespace pattern starts with solidus '/'

Note that if in the future we need to either customize regex pairs, or add a new kind of pairs, we will be able to do that by inventing syntax for the text before the starting '/', and that will be a backward-compatible change.